project_task_category/models/project_task_category.py

```diff
@@ -11,4 +11,5 @@ class ProjectTask(models.Model):
     categ_id = fields.Many2one(
         'project.category',
         'Category',
+        domain=[('code', '=', 'categ_id')]
```

Comment:
@dreispt  Another thing is that I think it's really hard to manage even when start with a clean DB 

I already I tried to do as you suggest before I create this PR and I think we add too much complexity for a simple problem that could add lot of management work for users and later if made a mistake will not work as is expected.

Imagine I have a set of categories without parenting. 

Then I want to have one as a set for `Root Task category` (category available on categ_id  AKA: Category)
I start to add as others categories as child of this root.
 
Finish with that I start with another set for `Root Project category` (category available on categ_ids AKA Tags).
But here is not possible to know if I select one category already present on another root, right? 
and because one2many relation could disappear from another root.

IMO this is a big problem of management and much worst for live environments with multiple tasks and categories created.